src/lambda/lambda.lisp

```diff
@@ -1 +1,268 @@
+
+(defmethod well-defp (ctx (tterm <stlc>))
+  (match-of stlc tterm
+    ((absurd)                 (and (well-defp       ctx (term tterm))
+                                   (geb.mixins:obj-equalp (type-of-term-w-fun ctx (term tterm))
+                                                          so0)))
+    (unit                     t)
+    ((left)                   (and (well-defp       ctx (term tterm))
+                                   (geb.mixins:obj-equalp (mcar (type-of-term-w-fun ctx tterm))
+                                                          (type-of-term-w-fun ctx (term tterm)))))
+    ((right)                  (and (well-defp       ctx (term tterm))
+                                   (geb.mixins:obj-equalp (mcadr (type-of-term-w-fun ctx tterm))
+                                                          (type-of-term-w-fun ctx (term tterm)))))
+    ((case-on on ltm rtm)     (let ((mcartoon (cons  (mcar (type-of-term-w-fun ctx on)) ctx))
+                                    (mcadrtoon (cons  (mcadr (type-of-term-w-fun ctx on)) ctx)))
+                                (and (well-defp mcartoon
+                                                ltm)
+                                     (well-defp mcadrtoon
+                                                rtm)
+                                     (well-defp ctx on)
+                                     (geb.mixins:obj-equalp (type-of-term-w-fun mcartoon ltm)
+                                                            (type-of-term-w-fun mcadrtoon rtm)))))
+    ((pair ltm rtm)           (let ((ttot (type-of-term-w-fun ctx tterm)))
+                                (and (well-defp ctx ltm)
+                                     (well-defp ctx rtm)
+                                     (geb.mixins:obj-equalp (mcar ttot)
+                                                            (type-of-term-w-fun ctx ltm))
+                                     (geb.mixins:obj-equalp (mcadr ttot)
+                                                            (type-of-term-w-fun ctx rtm)))))
+    ((fst)                    (and (well-defp       ctx (term tterm))
+                                   (geb.mixins:obj-equalp (type-of-term-w-fun ctx tterm)
+                                                          (mcar (type-of-term-w-fun ctx (term tterm))))))
+    ((snd)                    (and (well-defp       ctx (term tterm))
+                                   (geb.mixins:obj-equalp (type-of-term-w-fun ctx tterm)
+                                                          (mcadr (type-of-term-w-fun ctx (term tterm))))))
+    ((lamb tdom term)         (let ((ttot (type-of-term-w-fun ctx tterm)))
+                                (and (well-defp (cons tdom ctx) term)
+                                     (geb.mixins:obj-equalp (mcar ttot)
+                                                            tdom)
+                                     (geb.mixins:obj-equalp (mcadr ttot)
```

Comment:
you can just write `obj-equalp` `geb.mixins` is in scope due to the package using `geb.common`

it should help the lines not be so long. Please try to format things to around 80 characters long